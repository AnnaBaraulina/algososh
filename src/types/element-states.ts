export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}

export type ElementTypes = {
  letter: string;
  state: ElementStates;
 
}

export type SortTypes = {
  index: number;
  state: ElementStates;
}

export type CircleState = {
  modifiedIndex: number;
  changingIndex: number;
}

export enum CirclePosition  {
  head = 'head',
  tail = 'tail',
}