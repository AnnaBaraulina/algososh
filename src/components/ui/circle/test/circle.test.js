import { Circle } from "../circle";
import TestRenderer from 'react-test-renderer';
import { ElementStates } from "../../../../types/element-states";

describe('тест Circle', () => {
  test('Circle с буквами отрисован корректно', () => {
    const circle = TestRenderer.create(<Circle letter='Text'/>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  test('Circle без букв отрисован корректно', () => {
    const circle = TestRenderer.create(<Circle letter=""/>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  test('Circle с Head отрисован корректно', () => {
    const circle = TestRenderer.create(<Circle head='head'/>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  test('Circle с react-элементом в head отрисован корректно', () => {
    const circle = TestRenderer.create(<Circle head={<Circle/>}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  test('Circle с index отрисован корректно', () => {
    const circle = TestRenderer.create(<Circle index={1}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  test('Circle с пропом isSmall отрисован корректно', () => {
    const circle = TestRenderer.create(<Circle isSmall/>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  test('Circle в состоянии Default отрисован корректно', () => {
    const circle = TestRenderer.create(<Circle state={ElementStates.Default}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });
  test('Circle в состоянии Modified отрисован корректно', () => {
    const circle = TestRenderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });
})