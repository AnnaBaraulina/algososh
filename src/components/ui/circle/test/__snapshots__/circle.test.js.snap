// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тест Circle Circle без букв отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тест Circle Circle в состоянии Default отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тест Circle Circle в состоянии Modified отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тест Circle Circle с Head отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тест Circle Circle с index отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`тест Circle Circle с react-элементом в head отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тест Circle Circle с буквами отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Text
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`тест Circle Circle с пропом isSmall отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
